package/libskylark/src/libskylark.c

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright (C) 2016 Swift Navigation Inc.
+ * Contact: Jacob McNamee <jacob@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <curl/curl.h>
+#include <libsbp/observation.h>
+#include <libsbp/navigation.h>
+#include <string.h>
+#include <stdio.h>
+//#include <sbp_settings.h>
+#include <sbp_zmq.h>
+
+#include "libskylark.h"
+
+/**
+ *  Utilities
+ */
+
+// TODO (mookerji): Repeated here as is everwhere else!
+static int file_read_string(const char *filename, char *str, size_t str_size)
+{
+  FILE *fp = fopen(filename, "r");
+  if (fp == NULL) {
+    log_error("error opening %s\n", filename);
+    return -1;
+  }
+  bool success = (fgets(str, str_size, fp) != NULL);
+  fclose(fp);
+  if (!success) {
+    log_error("error reading %s\n", filename);
+    return -1;
+  }
+  return 0;
+}
```

Comment:
Yeah, noted. Will fix once I get a sense of where jacob's PRs are w.r.t. settings handling.